google-cloud-storage/src/main/java/com/google/cloud/storage/Notification.java

```diff
@@ -0,0 +1,313 @@
+  }
+
+  /** Returns the desired content of the Payload. */
+  public PayloadFormat getPayloadFormat() {
+    return payloadFormat;
+  }
+
+  /** Returns the object name prefix for which this notification configuration applies. */
+  public String getObjectNamePrefix() {
+    return objectNamePrefix;
+  }
+
+  /**
+   * Returns HTTP 1.1 Entity tag for the notification.
+   *
+   * @see <a href="http://tools.ietf.org/html/rfc2616#section-3.11">Entity Tags</a>
+   */
+  public String getEtag() {
+    return etag;
+  }
+
+  /**
+   * Returns the list of event types that this notification will apply to. If empty, notifications
+   * will be sent on all event types.
+   *
+   * @see <a href="https://cloud.google.com/storage/docs/cross-origin">Cross-Origin Resource Sharing
+   *     (CORS)</a>
+   */
+  public List<String> getEventTypes() {
+    return eventTypes;
+  }
+
+  /**
+   * Returns the list of additional attributes to attach to each Cloud PubSub message published for
+   * this notification subscription.
+   *
+   * @see <a href="https://cloud.google.com/storage/docs/access-control#About-Access-Control-Lists">
+   *     About Access Control Lists</a>
+   */
+  public Map<String, String> getCustomAttributes() {
+    return customAttributes;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(getTopic());
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj == this
+        || obj != null
+            && obj.getClass().equals(Notification.class)
+            && Objects.equals(toPb(), ((Notification) obj).toPb());
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this).add("topic", getTopic()).toString();
+  }
+
+  com.google.api.services.storage.model.Notification toPb() {
+    com.google.api.services.storage.model.Notification notificationPb =
+        new com.google.api.services.storage.model.Notification();
+    notificationPb.setId(generatedId);
+    notificationPb.setEtag(etag);
+    if (customAttributes != null) {
+      notificationPb.setCustomAttributes(customAttributes);
+    }
+    if (eventTypes != null) {
+      notificationPb.setEventTypes(eventTypes);
+    }
+    if (objectNamePrefix != null) {
+      notificationPb.setObjectNamePrefix(objectNamePrefix);
+    }
+    if (payloadFormat != null) {
+      notificationPb.setPayloadFormat(payloadFormat.toString());
+    } else {
+      notificationPb.setPayloadFormat(PayloadFormat.NONE.toString());
+    }
+    notificationPb.setSelfLink(selfLink);
+    notificationPb.setTopic(topic);
+
+    return notificationPb;
+  }
+
+  /**
+   * Creates a {@code Notification} object for the provided topic name.
+   *
+   * @param topic The name of the topic. It must have the format
+   *     "projects/{project}/topics/{topic}".
+   */
+  public static Notification of(String topic) {
+    PATH_TEMPLATE.validatedMatch(topic, "topic name must be in valid format");
+    return newBuilder(topic).build();
+  }
+
```

Comment:
Please consider creating a private static function to validate the format. This function will be called each time the topic is set.

Reply:
same thing we are doing here [see](https://googleapis.github.io/api-common-java/1.1.0/apidocs/com/google/api/pathtemplate/PathTemplate.html) and i don't think so we need to create an extra function here.

Reply:
I meant here that instead of calling `PATH_TEMPLATE.validatedMatch(topic, "topic name must be in valid format");` multiple times it's better to introduce:

private static void checkTopicFormat(String topic) {
   PATH_TEMPLATE.validatedMatch(topic, "topic name must be in valid format");
}

and call this function.